File: backend/Models/Product.cs
namespace backend.Models;

public class Product
{
    public int ProductId { get; set; }
    public required string ProductName { get; set; }
    public required string Description { get; set; }
    public decimal Price { get; set; }
    public int Stock { get; set; }
    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;
    public List<ProductImage> Images { get; set; } = new();
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/product.ts
export type ProductImage = {
  productImageId: number;
  productImageUrl: string;
  displayOrder: number;
}

export type Product = {
  productId: number;
  productName: string;
  description: string;
  price: number;
  stock: number;
  images: ProductImage[];
  categoryName: string;
  createdAt: string;
};
